### PR TITLE
Fix playlist order

### PR DESCRIPTION
**Pull request by quyet-v**
Changes
Modified logic to calculate adjusted index for playlist movement.

Issues
https://github.com/jellyfin/jellyfin/issues/13651

### CODE REVIEW

**Comment by quyet-v** [comment]
This should fix up some edge cases

File: tests/Jellyfin.Server.Implementations.Tests/Playlists/PlaylistManagerTests.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+using Emby.Server.Implementations.Playlists;
+using Xunit;
+
+namespace Jellyfin.Server.Implementations.Tests.Playlists
```
Comment (Bond-009) [inline tests/Jellyfin.Server.Implementations.Tests/Playlists/PlaylistManagerTests.cs:4]: Use file scoped namespaces for new files

**Review by Shadowghost — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Review by crobibero — APPROVED** [review]
